pkg/cache/placeholder.go

```diff
@@ -92,11 +92,12 @@ func newPlaceholder(placeholderName string, app *Application, taskGroup interfac
 			Name:      placeholderName,
 			Namespace: app.tags[constants.AppTagNamespace],
 			Labels: utils.MergeMaps(taskGroup.Labels, map[string]string{
-				constants.LabelApplicationID:   app.GetApplicationID(),
-				constants.LabelQueueName:       app.GetQueue(),
 				constants.LabelPlaceholderFlag: "true",
```

Comment:
If we prefer to use annotations, we may need to set the value in annotations.

Reply:
Hi @FrankYang0529,
Base on my understanding, I've set queueName and AppID to annotation in admission controller.
And have set annotation value is newPlaceholder.  [(Code)](https://github.com/apache/yunikorn-k8shim/pull/677/files#diff-2c7c04bd874ac06a9c003357d6129e83fff3e161e92f3a46cbcd1be545219f75R97-R100)

In the admission controller function pkg/admission/admission_controller#updateApplicationInfo,
I've moved the labels below to annotations. So the newPlaceholder should be able to fetch it.
- applicationId
- queue
- disableStateAware

Please kindly let me know if I have any misunderstanding.

Reply:
Sorry for the unclear comment. I thought we can replace `constants.LabelPlaceholderFlag` with `constants.AnnotationPlaceholderFlag` in annotations, so we can just use `Labels: taskGroup.Labels` here.

Reply:
Thanks @FrankYang0529's comment.

Actually I'm not sure whether we should move placeholder to annotation. 
In some code of e2e test. We use label selector to count how many placeholder pod is running.
- https://github.com/apache/yunikorn-k8shim/blob/5fe1b952bd5eb88bf21f49914216b9d49312cec0/test/e2e/gang_scheduling/gang_scheduling_test.go#L147
- https://github.com/apache/yunikorn-k8shim/blob/5fe1b952bd5eb88bf21f49914216b9d49312cec0/test/e2e/framework/helpers/k8s/k8s_utils.go#L975-L977

Label selector is native support by Kubernetes, but annotation is not.
If I moved placeholder to annotation, I have to list all pod and filter by each pod's annotation value to get placeholder pod. 
This change might be ok in e2e test, but will we require label selector for placeholder pod in the future?

Reply:
I check the code again and found that we already set `constants.AnnotationPlaceholderFlag`.

https://github.com/apache/yunikorn-k8shim/blob/5fe1b952bd5eb88bf21f49914216b9d49312cec0/pkg/cache/placeholder.go#L53-L56

@wilfred-s, do you think we also need to set `constants.LabelPlaceholderFlag` for label selector purpose? If yes, I think we may need to add some comments here. Thank you.

Reply:
In the view of back-forward compatibility. I will keep PlaceholderFlag's label and annotation in placeholder pod.

For the same reason, I should move applicationId/queue back to labels in placeholder pod, 
@wilfred-s Do I have the correct understanding about the newPlaceHolder change.

Reply:
To minimum the change of this PR and keep back-forward compatibility.
Just add label applicationId/queue back to placeholder pod. 26dc73107f5206ad7cdd021e1af8635d9efe803d